test/test_pipeline/test_classification.py

```diff
@@ -425,102 +553,140 @@ def _test_configurations(self, configurations_space, make_sparse=False,
-        self.assertEqual(
-            cs.get_hyperparameter('classifier:__choice__'),
-            CategoricalHyperparameter('classifier:__choice__', ['libsvm_svc']),
-            )
+        """Test the configuration space when using include and exclude
+
+        Expects
+        -------
+        * Including a classifier choice has pipeline give back matching choice
+        * Excluding a classifier choice means it won't show up in the hyperparameter space
+        * Including a feature preprocessor has pipeline give back matching choice
+        * Excluding a feature preprocessor means it won't show up in the hyperparameter space
+        """
+        # include a classifier choice
+        pipeline = SimpleClassificationPipeline(include={'classifier': ['libsvm_svc']})
+        cs = pipeline.get_hyperparameter_search_space()
 
-        cs = SimpleClassificationPipeline(exclude={'classifier': ['libsvm_svc']}).\
-            get_hyperparameter_search_space()
+        expected = CategoricalHyperparameter('classifier:__choice__', ['libsvm_svc'])
+        returned = cs.get_hyperparameter('classifier:__choice__')
+        self.assertEqual(returned, expected)
+
+        # exclude a classifier choice
+        pipeline = SimpleClassificationPipeline(exclude={'classifier': ['libsvm_svc']})
+        cs = pipeline.get_hyperparameter_search_space()
         self.assertNotIn('libsvm_svc', str(cs))
 
-        cs = SimpleClassificationPipeline(
-            include={'feature_preprocessor': ['select_percentile_classification']}).\
-            get_hyperparameter_search_space()
-        fpp1 = cs.get_hyperparameter('feature_preprocessor:__choice__')
-        fpp2 = CategoricalHyperparameter(
-            'feature_preprocessor:__choice__', ['select_percentile_classification'])
-        self.assertEqual(fpp1, fpp2)
+        # include a feature preprocessor
+        pipeline = SimpleClassificationPipeline(
+            include={'feature_preprocessor': ['select_percentile_classification']}
+        )
+        cs = pipeline.get_hyperparameter_search_space()
+
+        returned = cs.get_hyperparameter('feature_preprocessor:__choice__')
+        expected = CategoricalHyperparameter(
+            'feature_preprocessor:__choice__',
+            ['select_percentile_classification']
+        )
+        self.assertEqual(returned, expected)
 
-        cs = SimpleClassificationPipeline(
+        # exclude a feature preprocessor
+        pipeline = SimpleClassificationPipeline(
             exclude={'feature_preprocessor': ['select_percentile_classification']}
-        ).get_hyperparameter_search_space()
+        )
+        cs = pipeline.get_hyperparameter_search_space()
         self.assertNotIn('select_percentile_classification', str(cs))
 
     def test_get_hyperparameter_search_space_preprocessor_contradicts_default_classifier(self):
-        cs = SimpleClassificationPipeline(
+        """TODO: Not sure what this test does"""
```

Comment:
Somewhere there's a list of default classifiers and it starts with a random forest. However, the random forest is illegal after the densifier because it could handle sparse data by itself and densifying it would be a waste of resources. This test checks that the random forest is therefore not the default.

Reply:
Updated test doc, from what I could gather, there doesn't seem to be an explicit list specifying the default order. However there's a few while loops with try excepts which are popping values from dictionaries so I assume this is it. I documented it as much as I could